### DISCUSSION TITLE
Object types in views

### DISCUSSION THREAD

**Discussion by hjaltij** [matt8707/ha-fusion#119] [General]
First of all, this project is looking great.

Maybe I'm misunderstanding something but why does the sidebar offer a lot more options for object types than the main area / views?

I only see Camera and Button for the views but the sidebar has a lot more to choose from.

Thanks!

**Comment by janiskelemen** [comment]
I think the sidebar is mainly for displaying data and details about entities. The main section is for controlling your entities (toggle lights etc) and details are shown in modals when you click on the buttons. I guess there will be other types in the future but the focus right now is to add basic support for most common entities first and add more advanced things later.
@matt8707 please correct me if I'm wrong here.

**Reply by hjaltij** [reply]
Feels a bit bakcwards to me as there’s very limited space in the sidebar. I’d like some of my views to have graphs but I see the appeal of the simplicity with allowing only controls in the main area.

It would be great if those controls could be expanded like the climate entity so you can interact it without opening a modal.

**Comment by hjaltij** [comment]
Don’t get me wrong, I understand that is project is very young and I’m not expecting this functionality to be there yet. Mainly trying to understand the design and focus of the project.

**Reply by janiskelemen** [reply]
Makes sense, maybe @matt8707 can share what we can expect from this project and what is not in scope of it.

**Reply by matt8707** [reply]
> I guess there will be other types in the future but the focus right now is to add basic support for most common entities first and add more advanced things later.

Yeah, that's it really

**Reply by nagtegaal** [reply]
It would be nice to have some sort of overview/roadmap to see what the goal will be for this project and way to it.
I really think it has great potential, although - at least to me - some decisions at this point are not very clear to me what needs to be accomplished in the end.

**Reply by ysmilda** [reply]
What would make sense to me is that any `object` can be put in any `view` (`sidebar` or `main`). I'm not sure how it is set up now but from the project source it seems that they have a different implementation. Is there any limitation to just make them the same? That would make it much more versatile.

**Reply by hjaltij** [reply]
@ysmilda Right, that's what was confusing me, why they were made to be different.

**Reply by ysmilda** [reply]
![image](https://github.com/matt8707/ha-fusion/assets/19775528/987be17d-0387-4d42-a718-04ca58682849)

This is what I got working (crudely) in an hour of tinkering. I see possibilities where any object type is supported anywhere. This way there is only maintenance on one set of object while allowing endless customisation. 

The way I envision it is that the sidebar can act as a quick overal view of the home while the main page with the views gives detailed insights.

@matt8707 Is this in line with your vision? If so I'd be willing to help make this change happen.

**Reply by matt8707** [reply]
It isn't a technical obstacle. The plan is to create one 'sidebar graph' and one 'main graph' because they will be slightly different. Creating two will make development easier. Each should have its own distinct look and functionality. Once these are in place, we can optimize them, compare the differences and merge them into a one. Yes, the plan is that we can freely move them between the sidebar and main areas, enabling them to change reactively. Remeber that the current state of this project is in pre-beta. To help with development, you could port sidebar items to the main view instead.

**Reply by ysmilda** [reply]
The way I see it is that this will be creating a lot of extra work and implementing features twice in the future.

To me it would make sense to have these difference under a toggle to leave it up to the user what they want to display where. This way the general setup of the interface could be as simple as:
 - Enable or disable the sidebar in the settings
 - Add object
 - Select type, entity, look and feel
 - Drag to desired location (either sidebar or view)

This way there is less maintenance and more flexibility to let the user be creative with it.

**Reply by matt8707** [reply]
Yes, as I mentioned, that is the plan. However, for the sake of development speed, it is currently segregated for prototyping.

**Reply by ysmilda** [reply]
I didn't quite get that from your earlier reply, but it's great that that's the vision. As for development now; wouldn't it be more beneficial to start implementing this now instead of porting the sidebar to the main view? 

This is quite a big change with a big reach which will probably become more difficult as time goes on. It seems faster to rewrite it to a single `item` type now and than add all the upcoming changes.

**Reply by matt8707** [reply]
Yeah, I get what you're saying, and you're welcome to try, but I will not merge it if it introduces more issues than it solves.

**Reply by Th3AL0** [reply]
Was there any further decision in regards to this topic?

**Comment by Genemills** [comment]
I guess not!
